### PR TITLE
Datatable ts

### PR DESCRIPTION
**Pull request by xwatkins**
### Reference to issue
Convert `protvista-datatable` to TypeScript 

### Tests / Styleguide
 - [x] Tests pass
 - [x] Styleguide

### CODE REVIEW

File: packages/protvista-datatable/src/protvista-datatable.ts [hunk 1]
```diff
@@ -1,26 +1,79 @@
...
+
+import { ProtvistaLoadEvent } from "./types/events";
+import { ProtvistaManager } from "./types/manager";
+
 import styles from "./styles";
 
-class ProtvistaDatatable extends LitElement {
+type DataTableDatum = {
+  start?: number;
+  begin?: number;
```
Comment (dlrice) [inline packages/protvista-datatable/src/protvista-datatable.ts:12]: I think here you need at least one of these? If so, type-fest has a [require-at-least-one](https://github.com/sindresorhus/type-fest/blob/main/source/require-at-least-one.d.ts) that could enforce that.

File: packages/protvista-datatable/src/protvista-datatable.ts [hunk 2]
```diff
@@ -48,32 +101,40 @@ class ProtvistaDatatable extends LitElement {
...
-    ) {
+  eventHandler(e: MouseEvent): void {
+    const target = e.target as HTMLElement;
+    if (!target.closest("protvista-datatable") && !target.closest(".feature")) {
       this.selectedid = null;
     }
   }
 
-  static get properties() {
+  static get properties(): { [key: string]: any } {
```
Comment (dlrice) [inline packages/protvista-datatable/src/protvista-datatable.ts:121]: Is it possible to eliminate the `any` here?

File: packages/protvista-datatable/src/protvista-datatable.ts [hunk 3]
```diff
@@ -1,26 +1,91 @@
...
+  LitElement,
+  html,
+  TemplateResult,
+  CSSResult,
+  PropertyDeclarations,
+} from "lit-element";
 import { v1 } from "uuid";
 import { ScrollFilter } from "protvista-utils";
-/* eslint-disable import/extensions, import/no-extraneous-dependencies */
+// Not sure why, but eslint doesn't see it...
```
Comment (dlrice) [inline packages/protvista-datatable/src/protvista-datatable.ts:10]: With the following in `.eslintrc.json`:
```
  "settings": {
    "import/resolver": {
      "node": {
        "extensions": [".js", ".ts",".d.ts"]  // <---with ".d.ts" now
      }
    }
  }
```
the linter shouldn't complain anymore.

Comment (xwatkins) [reply packages/protvista-datatable/src/protvista-datatable.ts:10]: You, sir, are a gentleman and a scholar. I'd assumed ".ts" would have covered ".d.ts", but no.

**Review by dlrice — APPROVED** [review]

**Pull request merged**